### ISSUE TITLE
Get /pages throws a 500 if there is a folder in Site Pages

### ISSUE THREAD

**Issue by ghost** [microsoftgraph/microsoft-graph-docs#4928]
Hi all,

I'm trying to list all pages from my SharePoint site via the GET /pages but I'm getting a 400 with the following message: "Provided identifier is malformed - site collection id is not valid". 

It's weird because when I do /sites/contoso.sharepoint.com it works fine as well as when I do /sites/contoso.sharepoint.com/lists (using my hostname of course), the issue is only with the /pages endpoint.

When trying to pass the whole ID (Hostname,SPsite-id,SPweb-id) on the Request URL, I get a 500 with the following message "General exception while processing".

Any idea on what could be happening? The documentation provided on this page doesn't seems to be accurate.

Thanks in advance :)

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 0922533d-b396-4bd5-8258-7ce6f2a5b4b0
* Version Independent ID: d1719bc1-65aa-d4e6-24d1-524f35c4ee21
* Content: [List the pages in a site - Microsoft Graph beta](https://docs.microsoft.com/en-us/graph/api/sitepage-list?view=graph-rest-beta&tabs=cs)
* Content Source: [api-reference/beta/api/sitepage-list.md](https://github.com/microsoftgraph/microsoft-graph-docs/blob/master/api-reference/beta/api/sitepage-list.md)
* Product: **sharepoint**
* Technology: **microsoft-graph**
* GitHub Login: @rahmit
* Microsoft Alias: **rahmit**

**Comment by rahmit** [comment]
Thanks for reporting this! I do have a repro and will investigate.

**Comment by amanda-trigueiro** [comment]
Hi @rahmit, just an update with this issue.

I could successfully GET /pages now. The issue was that I had a Template folder inside my Site Pages list. When I deleted the folder, I got a 200 OK. Created the folder again, got a 500 with the "General exception while processing" message. Deleted again, 200 OK.

Plus, the call only works passing the full id {Hostname,SPsite-id,SPweb-id}; Using the hostname only throws a 400 (Provided identifier is malformed - site collection id is not valid), even if there is no folder on the Site Pages list.

Hope it helps. Will change the title of the issue to be more accurate with the problem.

Thanks,
Amanda.
